src/OpenTelemetry.Api/Trace/ActivityExtensions.cs

```diff
@@ -73,5 +74,32 @@ public static Status GetStatus(this Activity activity)
 
             return status;
         }
+
+        /// <summary>
+        /// Sets the kind of activity execution.
+        /// </summary>
+        /// <param name="activity">Activity instance.</param>
+        /// <param name="kind">Activity execution kind.</param>
+        public static void SetKind(this Activity activity, ActivityKind kind)
+        {
+            if (activity == null)
+            {
+                throw new ArgumentNullException(nameof(activity));
+            }
+
+            SetKindProperty(activity, kind);
+        }
+
+#pragma warning disable SA1201 // Elements should appear in the correct order
+        internal static Action<Activity, ActivityKind> SetKindProperty = CreateActivityKindSetter();
```

Comment:
Why internal for the two statics?

Reply:
Those don't need to be exposed. What are your thoughts?

Reply:
Basically what I'm asking is: Why not `private`?

Reply:
done!